include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,298 @@
+    //! @brief Adds an option to current callable
+    //!
+    //! If the option `o` is supported by one of the specifications `Spec`, build a new @callable that will behave as
+    //! an instance of `Tag` except the option `o` is passed inside an eve::options as an additional parameter.
+    //!
+    //! Does not participate in overload resolution if `o` is not support by any option specifications.
+    //!
+    //! @param o Option to add to current's @callable setup
+    //! @return A eve::decorated_callable behaving as `Tag` but with the options `o` set.
+    //==================================================================================================================
+    template<typename O>
+    EVE_FORCEINLINE auto operator[](O const& o) const
+    -> decltype( detail::decorated_fn<Tag,decltype(process_option(defaults(), o)),Specs...>(process_option(defaults(), o)) )
+    {
+      auto opts = process_option(defaults(), o);
+      return detail::decorated_fn<Tag,decltype(opts),Specs...>(opts);
+    }
+  };
+
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct conditional_option
+  //!   @brief Option specification for decoration via conditional value and expressions
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   eve::conditional_option is an option specification that can be used when defining a eve::callable type to make
+  //!   it supports decoration via `bool`, eve::logical, any eve::conditional_expr or eve::relative_conditional_expr.
+  //!
+  //!   @groupheader{Example}
+  //!
+  //!   @godbolt{doc/traits/callable_supports.cpp}
+  //! @}
+  //====================================================================================================================
+  struct conditional_option
+  {
+    auto process_option(auto const& base, rbr::concepts::exactly<condition_key> auto opt) const
+    {
+      return options{rbr::merge(options{opt}, base)};
+    }
+
+    template<std::same_as<bool> O>
+    auto process_option(auto const& base, O opt) const
+    {
+      // Just delay the evaluation of the type by injecting some templates
+      using type = std::conditional_t<std::same_as<bool,O>,std::uint8_t,O>;
+      return process_option(base, condition_key = if_(logical<type>(opt)) );
+    }
+
+    auto process_option(auto const& base, eve::logical_value auto opt) const
+    {
+      return process_option(base, condition_key = opt);
+    }
+
+    auto process_option(auto const& base, eve::conditional_expr auto opt) const
+    {
+      return process_option(base, condition_key = opt);
+    }
+
+    /// Default settings of eve::conditional is eve::ignore_none
+    EVE_FORCEINLINE static constexpr auto defaults() noexcept { return options{condition_key = ignore_none};  }
+  };
```

Comment:
hmm, do we want to just inherit relative_condition_option?

Reply:
I was thinking about it but wasn't sure.